src/component/registry/bpmn-model-registry.ts

```diff
@@ -78,12 +78,14 @@ export class BpmnModelRegistry {
 }
 
 function toRenderedModel(bpmnModel: BpmnModel): RenderedModel {
-  const collapsedSubProcessIds: string[] = bpmnModel.flowNodes
-    .filter(shape => {
-      const bpmnElement = shape.bpmnElement;
-      return ShapeUtil.isSubProcess(bpmnElement.kind) && (bpmnElement as ShapeBpmnSubProcess).markers.includes(ShapeBpmnMarkerKind.EXPAND);
-    })
-    .map(shape => shape.bpmnElement.id);
+  const collapsedSubProcessIds = new Set(
+    bpmnModel.flowNodes
+      .filter(shape => {
+        const bpmnElement = shape.bpmnElement;
+        return ShapeUtil.isSubProcess(bpmnElement.kind) && (bpmnElement as ShapeBpmnSubProcess).markers.includes(ShapeBpmnMarkerKind.EXPAND);
+      })
+      .map(shape => shape.bpmnElement.id),
+  );
```

Comment:
**thought:** personal quote, I don't understand why this changes have been done: why introducing a Set here?
Need to test locally to see if this is because of an eslint error. Otherwise, this is unrelated to the topic of this PR

Reply:
![](https://github.com/process-analytics/bpmn-visualization-js/assets/4921914/bc3e90c4-8eb8-4177-b172-4733309c7332)